source/utilities/imgcat

```diff
@@ -86,6 +86,29 @@ while [ $# -gt 0 ]; do
     -p|--p|--print)
         print_filename=1
         ;;
+	-u|--u|--url)
+        if curl --output /dev/null --silent --head --fail "$2"; then # check if the url is downloadable
+			# create a file with unique file name
+			file=$(mktemp -t curlimg) || { echo "Failed creating file"; return; }
+			# download file and save it into the unique file name just created
+    		curl -s "$2" > $file || { echo "Failed to curl file"; return; }
+			# reset argument of url to the corresponding unique file
+            has_stdin=f
+            print_image "$file" 1 "$(base64 < "$file")" "$print_filename"
+			# delete the downloaded file
+			rm $file
+			# only modify second argument in order to keep running option "-u"
+			# shift will remove the first arguement "-u"
+			set -- ${@:1:1} "-u" ${@:3}
```

Comment:
Please explain exactly what this command does. I've never seen this syntax before and it's basically impossible to google for.